src/tools/nix-dev-shell/shell.nix

```diff
@@ -16,10 +16,15 @@ pkgs.mkShell {
     pkgs.nix
     pkgs.glibc.out
     pkgs.glibc.static
+    pkgs.rust-analyzer
     x
     # Get the runtime deps of the x wrapper
   ] ++ lists.flatten (attrsets.attrValues env);
 
+  shellHook = ''
+    export PATH=$(pwd)/build/host/stage0/bin:$PATH
```

Comment:
what is happening here?

Reply:
I don't want stage0 rustc in my path 😨, I want rustup rustc in my path which I can use properly to use my linked toolchains and everything

Reply:
Is there anything stopping you from just using rustup in the shell?

This will break using rustc directly so that seems really bad

Reply:
nothing is technically stopping me, I just don't want to use rustup ^^'

either way, I removed this. I fixed the problem on my side by adding the following to my helix config: 
```toml
[language-server.rust-analyzer.environment]
RUSTC = "/home/wffl/prog/rust-r/build/host/stage0/bin/rustc"
CARGO = "/home/wffl/prog/rust-r/build/host/stage0/bin/cargo"
```